lib/lookbook/preview_parser.rb

```diff
@@ -6,14 +6,14 @@ def initialize(paths, tags = nil)
       @paths = paths
       @after_parse_callbacks = []
       @after_parse_once_callbacks = []
-      @parsing = false
+      @parsing = Mutex.new
 
       define_tags(tags)
       YARD::Parser::SourceParser.after_parse_list { run_callbacks }
     end
 
     def parse(files = nil, &block)
-      unless @parsing
+      @parsing.synchronize do
         @parsing = true
```

Comment:
@joshuay03 I think this line needs to be removed doesn't it? Otherwise it overwrites the mutex?

Reply:
Oops yes. Nice catch!

Reply:
Fixed 👍🏽